diff -urw vitro-rel-1.4.1/webapp/src/edu/cornell/mannlib/vitro/webapp/controller/authenticate/BasicAuthenticator.java ../vitro-rel-1.4.1/webapp/src/edu/cornell/mannlib/vitro/webapp/controller/authenticate/BasicAuthenticator.java
--- vitro-rel-1.4.1/webapp/src/edu/cornell/mannlib/vitro/webapp/controller/authenticate/BasicAuthenticator.java	2012-02-07 23:41:01.000000000 +1100
+++ ../vitro-rel-1.4.1/webapp/src/edu/cornell/mannlib/vitro/webapp/controller/authenticate/BasicAuthenticator.java	2013-07-12 15:09:42.000000000 +1000
@@ -303,7 +303,7 @@
 
 		ServletContext servletContext = session.getServletContext();
 		WebappDaoFactory wadf = (WebappDaoFactory) servletContext
-				.getAttribute("webappDaoFactory");
+				.getAttribute("assertionsWebappDaoFactory");
 		if (wadf == null) {
 			log.error("no WebappDaoFactory");
 			return null;
